### PR TITLE
Fix

### PR DESCRIPTION
**Pull request by Laci1337**

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -28,20 +28,22 @@ public TorpedoStore(int numberOfTorpedos){
     }
   }
 
+  private Random generator = new Random();
```
Comment (imrebalint17) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:31]: Well done

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -28,20 +28,22 @@ public TorpedoStore(int numberOfTorpedos){
...
+
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (imrebalint17) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: Nice

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -28,20 +28,22 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+   
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (imrebalint17) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:46]: Well made

**Review by imrebalint17 — CHANGES_REQUESTED** [review]

**Review by Laci1337 — COMMENTED** [review]
Kösz

**Review by imrebalint17 — APPROVED** [review]

**Comment by Laci1337** [comment]
El lett fogadva

**Pull request merged**